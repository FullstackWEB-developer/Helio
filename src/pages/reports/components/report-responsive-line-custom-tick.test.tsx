import dayjs from 'dayjs';
import duration from 'dayjs/plugin/duration';
import utc from 'dayjs/plugin/utc';
import {unmountComponentAtNode} from 'react-dom';
import {render} from '@testing-library/react';
import TestWrapper from '@shared/test-utils/test-wrapper';
import ReportResponsiveLineCustomTick from '@pages/reports/components/report-responsive-line-custom-tick';

describe("Report Responsive Line Custom Tick Test", () => {
    let container: HTMLDivElement | null;
      beforeEach(async () => {
        dayjs.extend(duration);
        dayjs.extend(utc);
        container = document.createElement("div");
        document.body.appendChild(container);
    });

    afterEach(() => {
        if (container) {
            unmountComponentAtNode(container);
            container.remove();
            container = null;
        }
    });

    it("renders correctly", async () => {
        const {asFragment} = render(<TestWrapper>
            <ReportResponsiveLineCustomTick
                tick={{
                    value: '9:00 AM'
                }}
                isTime={true}/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders time if isTime == true", async () => {
        const {asFragment, findByText} = render(<TestWrapper >
            <ReportResponsiveLineCustomTick
                tick={{
                    value: '9:00 AM'
                }}
                isTime={true}/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
        expect(findByText('9 AM')).not.toBeNull();
    });

    it("renders date if isTime != true", async () => {
        const {asFragment, findByText} = render(<TestWrapper >
            <ReportResponsiveLineCustomTick
                tick={{
                    value: 'Aug 01'
                }}
                isTime={true}/>
        </TestWrapper>);
        expect(asFragment()).toMatchSnapshot();
        expect(findByText('Aug 01')).not.toBeNull();
    });

})
