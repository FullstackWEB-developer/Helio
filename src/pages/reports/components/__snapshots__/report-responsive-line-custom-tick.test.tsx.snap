// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report Responsive Line Custom Tick Test renders bot reports correctly 1`] = `
<DocumentFragment>
  <svg>
    <g
      transform="translate(undefined,NaN)"
    >
      <text
        dominant-baseline="middle"
        style="font-size: 0px;"
        text-anchor="middle"
      >
        Invalid Date
      </text>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`Report Responsive Line Custom Tick Test renders correctly 1`] = `
<DocumentFragment>
  <svg>
    <g
      transform="translate(undefined,NaN)"
    >
      <text
        dominant-baseline="middle"
        style="font-size: 0px;"
        text-anchor="middle"
      >
        Invalid Date
      </text>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`Report Responsive Line Custom Tick Test renders date if isTime != true 1`] = `
<DocumentFragment>
  <svg>
    <g
      transform="translate(undefined,NaN)"
    >
      <text
        dominant-baseline="middle"
        style="font-size: 0px;"
        text-anchor="middle"
      >
        9 PM
      </text>
    </g>
  </svg>
</DocumentFragment>
`;

exports[`Report Responsive Line Custom Tick Test renders time if isTime == true 1`] = `
<DocumentFragment>
  <svg>
    <g
      transform="translate(undefined,NaN)"
    >
      <text
        dominant-baseline="middle"
        style="font-size: 0px;"
        text-anchor="middle"
      >
        Invalid Date
      </text>
    </g>
  </svg>
</DocumentFragment>
`;
